src/functionaltests/java/com/ericsson/ei/subscriptions/query/FreestyleQueryAggregatedObjectsTestSteps.java

```diff
@@ -0,0 +1,154 @@
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+
+import com.ericsson.ei.mongodbhandler.MongoDBHandler;
+import com.ericsson.ei.utils.FunctionalTestBase;
+
+
+@Ignore
+@AutoConfigureMockMvc
+public class FreestyleQueryAggregatedObjectsTestSteps extends FunctionalTestBase {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreestyleQueryAggregatedObjectsTestSteps.class);
+    
+    private static final String aggregatedObjJsonPath = "src/test/resources/AggregatedDocument.json";
+    private static final String missedNotificationJsonPath = "src/test/resources/MissedNotification.json";
+
+    @Autowired
+    private MockMvc mockMvc;
+    private MvcResult mvcResult;
+    
+    @Autowired
+    private MongoDBHandler mongoDBHandler;
+    
+    @Value("${spring.data.mongodb.database}")
+    private String eiDatabaseName;
+    
+    @Value("${aggregated.collection.name}")
+    private String aggrCollectionName;
+    
+    @Value("${missedNotificationCollectionName}")
+    private String missedNotificationCollectionName;
+    
+    @Value("${missedNotificationDataBaseName}")
+    private String missedNotificationDatabaseName;
+
+    private String aggrObj;
+    private String missedNotificationObj;
+    
+    private ObjectMapper objMapper;
+    
+    public FreestyleQueryAggregatedObjectsTestSteps() {
+        objMapper = new ObjectMapper();
+
+        try {
+            aggrObj = FileUtils.readFileToString(new File(aggregatedObjJsonPath), "UTF-8");
+            missedNotificationObj = FileUtils.readFileToString(new File(missedNotificationJsonPath), "UTF-8");
+
+        } catch (IOException e) {
+            LOGGER.error("Failed to open test json files for test. Error message\n: " + e.getMessage());
+            e.printStackTrace();
+        }
+    }
+    
+    @Given("^Aggregated object2 is created$")
+    public void aggregated_object_is_created() throws Throwable {
+        LOGGER.debug("Creating aggregated object in MongoDb");
+        assertEquals(true, createDocumentInMongoDb(eiDatabaseName, aggrCollectionName, aggrObj));
+    }
+
+    @Then("^Perform valid freestyle query on newly created Aggregated object$")
+    public void perform_valid_query_on_newly_created_aggregated_object() throws Throwable {
+        final String expectedAggrId = "6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43";
+        final String entryPoint = "/query";
+        final String queryAggrObj = "{\"criteria\" :{\"aggregatedObject.id\" : \"" + expectedAggrId + "\"}}";
```

Comment:
please add a more complex query that uses several fields as criteria and one of these fields should be as deep as possible.

Reply:
Done.